src/server/editorServices.ts

```diff
@@ -2705,22 +2711,30 @@ namespace ts.server {
 
         openClientFileWithNormalizedPath(fileName: NormalizedPath, fileContent?: string, scriptKind?: ScriptKind, hasMixedContent?: boolean, projectRootPath?: NormalizedPath): OpenConfiguredProjectResult {
             const info = this.getOrCreateOpenScriptInfo(fileName, fileContent, scriptKind, hasMixedContent, projectRootPath);
-            const result = this.assignProjectToOpenedScriptInfo(info);
-            this.cleanupAfterOpeningFile();
+            const { defaultConfigProject, ...result } = this.assignProjectToOpenedScriptInfo(info);
+            this.cleanupAfterOpeningFile(defaultConfigProject);
             this.telemetryOnOpenFile(info);
             return result;
         }
 
-        private removeOrphanConfiguredProjects() {
+        private removeOrphanConfiguredProjects(toRetainConfiguredProjects: ConfiguredProject[] | ConfiguredProject | undefined) {
             const toRemoveConfiguredProjects = cloneMap(this.configuredProjects);
+            if (toRetainConfiguredProjects) {
+                if (isArray(toRetainConfiguredProjects)) {
+                    toRetainConfiguredProjects.forEach(retainConfiguredProject);
+                }
+                else {
+                    retainConfiguredProject(toRetainConfiguredProjects);
+                }

```

Comment:
Suggestion:

```ts
if (toRetainConfiguredProjects) {
  toArray(toRetainConfiguredProjects).forEach(retainConfiguredProject);
}
```

Reply:
Would rather not create array just to iterate and call function.